### ISSUE TITLE
FileReadLine 

### ISSUE THREAD

**Issue by mmikeww** [mmikeww/AHK-v2-script-converter#20]
can replace 

```autohotkey
FileReadLine, line, C:\My Documents\ContactList.txt, 5
```

with

```autohotkey
Loop, Read, C:\My Documents\ContactList.txt
{
   if (A_Index = 5)
   {
      line := A_LoopReadLine
      break
   }
}
```

or using file objects

```autohotkey
file := FileOpen("C:\My Documents\ContactList.txt", "r")
Loop 5
   line := file.ReadLine()
file.Close()
```

**Comment by mmikeww** [comment]
what about this from the v1 help file?

```autohotkey
Loop
{
    FileReadLine, line, C:\My Documents\ContactList.txt, %A_Index%
    if ErrorLevel
        break
    MsgBox, 4, , Line #%A_Index% is "%line%".  Continue?
    IfMsgBox, No
        return
}
```

that would mean something like this
```autohotkey
Loop
{
    i := A_Index
    Loop, Read, C:\My Documents\ContactList.txt
    {
       if (A_Index = i)
       {
          line := A_LoopReadLine
          break
       }
    }
    if ErrorLevel
        break
    MsgBox, 4, , Line #%A_Index% is "%line%".  Continue?
    IfMsgBox, No
        return
}
```

**Comment by dmtr99** [comment]
This is the correct conversion (not yet implemented)
if ErrorLevel is not present in the script, we can skip the setting of it.

try{
ErrorLevel := 0
Text_FileRead := FileRead("C:\Users\Dimitri\Documents\FTsPLog.txt")
line := StrSplit(Text_FileRead,"`n","`r")[A_Index]
} Catch{
line := ""
ErrorLevel := 1
}

**Comment by andymbody** [comment]
> This is the correct conversion (not yet implemented) if ErrorLevel is not present in the script, we can skip the setting of it.
> 
> try{ ErrorLevel := 0 Text_FileRead := FileRead("C:\Users\Dimitri\Documents\FTsPLog.txt") line := StrSplit(Text_FileRead,"`n","`r")[A_Index] } Catch{ line := "" ErrorLevel := 1 }

Since this has not been closed... based on the above suggested code... would this be welcome edit to script?

```
_FileReadLine(p) {
   global Indentation
   ; FileReadLine, OutputVar, Filename, LineNum
   ; Not really a good alternative, inefficient but the result is the same

   ; 2024-06-28, AMB Issue #20
   ; is this proper conversion?
   nextLine := '`r`n'   . Indentation
   lineTab  := nextLine . '`t'
   cmd :=       'try {'
            .   lineTab     . 'Global ErrorLevel := 0'
            .   lineTab     . p[1] . ' := StrSplit(FileRead(" p[2] "),`"``n`",`"``r`")[' P[3] ']'
            .   nextLine    . '} Catch {'
            .   lineTab     . 'LineText := ""'
            .   lineTab     . 'ErrorLevel := 1'
            .   nextLine    . '}'

   Return cmd     ;p[1] " := StrSplit(FileRead(" p[2] "),`"``n`",`"``r`")[" P[3] "]"
}
```

 I thought I would ask before submitting a PR.

**Comment by andymbody** [comment]
> This is the correct conversion (not yet implemented) if ErrorLevel is not present in the script, we can skip the setting of it.
> 
> try{ ErrorLevel := 0 Text_FileRead := FileRead("C:\Users\Dimitri\Documents\FTsPLog.txt") line := StrSplit(Text_FileRead,"`n","`r")[A_Index] } Catch{ line := "" ErrorLevel := 1 }

Since this has not been closed, might I suggest the following based on the post above. If this is acceptable, I can send a PR.

```
_FileReadLine(p) {
   global Indentation
   ; FileReadLine, OutputVar, Filename, LineNum
   ; Not really a good alternative, inefficient but the result is the same

;   Return p[1] " := StrSplit(FileRead(" p[2] "),`"``n`",`"``r`")[" P[3] "]"

   ; is this proper conversion?
   newLine  := '`r`n'   . Indentation
   lineTab  := newLine  . '`t'
   cmd :=       'try {'
            .   lineTab     . 'Global ErrorLevel := 0'
            .   lineTab     . p[1] . ' := StrSplit(FileRead(" p[2] "),`"``n`",`"``r`")[' P[3] ']'
            .   newLine     . '} Catch {'
            .   lineTab     . p[1] . ' := ""'
            .   lineTab     . 'ErrorLevel := 1'
            .   newLine     . '}'

   Return cmd
}
```

**Comment by andymbody** [comment]
Is this one complete or is there more to do?

**Comment by Banaanae** [comment]
I just forgot to close
You can use a [closing keyword](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue) to automatically close these issues, for example `fix #20` `close #20`

**Comment by andymbody** [comment]
Ok... thanks for the tip!
